spot_rl_experiments/spot_rl/utils/heuristic_nav.py

```diff
@@ -401,5 +401,91 @@ def search(
         return False, (None, None, None), rgb_img_vis
 
 
+def heuristic_object_navigation(
+    x: float,
+    y: float,
+    theta: float,
+    object_target: str,
+    image_search: ImageSearch = None,
+    save_cone_search_images: bool = True,
+    pull_back: bool = True,
+    skillmanager=None,
+):
+
+    if save_cone_search_images:
+        previously_saved_images = glob("imagesearch*.png")
+        for f in previously_saved_images:
+            os.remove(f)
+
+    if image_search is None:
+        image_search = ImageSearch(
+            corner_static_offset=0.5,
+            use_yolov8=False,
+            visualize=save_cone_search_images,
+        )
+
+    (x, y) = (
+        pull_back_point_along_theta_by_offset(x, y, theta, 0.2) if pull_back else (x, y)
+    )
+    print(f"Nav targets adjusted on the theta direction ray {x, y, np.degrees(theta)}")
+
+    skillmanager.nav(x, y, theta)
+    skillmanager.nav_controller.nav_env.enable_nav_by_hand()
+
+    spot: Spot = skillmanager.spot
+    spot.open_gripper()
+    gaze_arm_angles = deepcopy(skillmanager.pick_config.GAZE_ARM_JOINT_ANGLES)
+    spot.set_arm_joint_positions(np.deg2rad(gaze_arm_angles), 1)
+    time.sleep(1.5)
+    found, (x, y, theta), visulize_img = image_search.search(
+        object_target, *get_arguments_for_image_search(spot, 0)
+    )
+    rate = 20  # control time taken to rotate the arm, higher the rotation higher is the time
+    if not found:
+        # start semi circle search
+        angle_interval = 20
+        semicircle_range = np.arange(-90, 110, angle_interval)
```

Comment:
Can we keep semicircle range as function input? So in case someone wants to call this with varied search angles, they still can do so?

Also angle interval of heuristic search can be moved to config? We shouldn't have magic numbers